EXAMPLES.md

```diff
@@ -0,0 +1,202 @@
+## Request Options
+
+Fine-grained configuration can be provided on a per-request basis to enhance the request with specific query params, headers or to pass it a custom context.
+
+```go
+// Example
+userGrants, err := auth0API.Grant.List(
+	management.Context(ctx)
+	management.Header("MySpecialHeader","MySpecialHeaderValue")
+    management.Parameter("user_id", "someUserID"),
+    management.Parameter("client", "someClientID"),
+    management.Parameter("audience", "someAPIAudience"),
+)
+
+// Other helper funcs.
+management.Query()
+management.ExcludeFields()
+management.IncludeFields()
+management.Page()
+management.PerPage()
+management.IncludeTotals()
+management.Take()
+management.From()
+```
+
+## Pagination
+
+This SDK supports both offset and checkpoint pagination.
+
+### Page based pagination
+
+When retrieving lists of resources, if no query parameters are set using the `management.PerPage` and `Management.IncludeTotals` helper funcs, then the SDK will default to sending `per_page=50` and `include_totals=true`. 
+
+> **Note**
+> The maximum value of the per_page query parameter is 100.
+
+In order to paginate using the page based pagination, you can follow a pattern like below:
+
+```go
+var page int
+for {
+    clients, err := auth0API.Client.List(
+        management.Page(page),
+        management.PerPage(100),
+    )
+    if err != nil {
+        return err
+    }
+
+    // Accumulate here the results or check for a specific client.
+
+    if !clients.HasNext() {
+        break
+    }
+
+    page++
+}
+```
+
+### Checkpoint pagination
+
+Checkpoint pagination can be used when you wish to retrieve more than 1000 results from certain APIs. The APIs that support checkpoint based pagination are:
+
+* `Log.List` (`/api/v2/logs`)
+* `Organization.List` (`/api/v2/organizations`)
+* `Organization.Members` (`/api/v2/organizations/{id}/members`)
+* `Role.Users` (`/api/v2/roles/{id}/users`)
+
+In general to use checkpoint based pagination you should follow a pattern like below:
+
+```go
+var orgList *management.OrganizationList
+for {
+    if orgList == nil {
+        // For the first call, only pass the `take` query parameter, the API will then return a
+        // `Next` value that can be used for future requests,
+        orgList, err = auth0API.Organization.List(
+            management.Take(100),
+        )
+    } else {
+        // Pass the `next` and `take` query parameters now so that we can correctly paginate the
+        // organizations,
+        orgList, err = auth0API.Organization.List(
+            management.From(orgList.Next),
+            management.Take(100),
+        )
+    }
+
+    if err != nil {
+        log.Fatal("err :%w", err)
+    }
+
+    for _, org := range orgList.Organizations {
+        log.Printf("org %s", org.GetID())
+    }
+
+    // The `HasNext` helper func checks whether the API has informed us that there is more data to
+    // retrieve or not.
+    if !orgList.HasNext() {
+        break
+    }
+}
+```
```

Comment:
Hopefully we can improve on this in the future, and avoid requiring this boilerplate.

Reply:
Yeah I'm a big fan of the auto-pagination in [auth0-php](https://github.com/auth0/auth0-PHP/blob/main/UPGRADE.md#auto-pagination-support-available) and would love to explore that in here (and other SDKs)

Reply:
💯